opentreemap/opentreemap/integrations/inaturalist.py

```diff
@@ -0,0 +1,120 @@
+import dateutil.parser
+from datetime import timedelta
+
+import requests
+from django.conf import settings
+from django.db import connection
+
+from treemap.models import INaturalistObservation, Species
+
+base_url = "https://www.inaturalist.org"
+
+
+def get_inaturalist_auth_token():
+
+    payload = {
+        'client_id': settings.INATURALIST_APP_ID,
+        'client_secret': settings.INATURALIST_APP_SECRET,
+        'grant_type': 'password',
+        'username': settings.USERNAME,
+        'password': settings.PASSWORD
+    }
+
+    r = requests.post(
+        url="{base_url}/oauth/token".format(base_url=base_url),
+        data=payload
+    )
+    token = r.json()['access_token']
+    return token
+
+
+def create_observation(token, latitude, longitude):
+
+    headers = {'Authorization': 'Bearer {}'.format(token)}
+    params = {'observation': {
+        'observed_on_string': datetime.datetime.now().isoformat(),
+        'latitude': latitude,
+        'longitude': longitude
+    }
+    }
+
+    response = requests.post(
+        url="{base_url}/observations.json".format(base_url=base_url),
+        json=params,
+        headers=headers
+    )
+
+    return response.json()[0]
```

Comment:
Not for right now, but we will want to check that the status is 200 and it actually has an observation.